### DISCUSSION TITLE
'Arithmetic operation resulted in an overflow.' in csvReader.GetRecords<T>

### DISCUSSION THREAD

**Discussion by dpenkova** [JoshClose/CsvHelper#2140] [Q&A]
I get 'Arithmetic operation resulted in an overflow.'  error while trying to read a csv file using csvHepler.

Does anyone have idea how can I solve this?

Thanks

![image](https://user-images.githubusercontent.com/5197659/228217741-b8fcbd36-1dae-4d30-97ee-9a23785a65ba.png)

**Comment by JoshClose** [comment]
You'll need to post some code here so we can try and determine the issue.

**Comment by Guacam-Ole** [comment]
Had the same issue. Seems like CsvReader does not like big files (>2GB). This even happens when trying to parse one by one on the first csvReader.Read(). My suggestion is to split the CSV manually before parsing it

**Reply by JoshClose** [reply]
Can you give some code to reproduce?

**Comment by Guacam-Ole** [comment]
Sorry. I don't have the exact code anymore because we solved this by created smaller CSVs and send them through csvhelper.

But is simply was just the code from the "by hand" example
https://joshclose.github.io/CsvHelper/examples/reading/reading-by-hand/

the only differencce was having a csvconfig with custom header-rules, so using a slight different constructor of CSVReader when compared to the example 

(The other ways importing CSV have the same issue but we just hoped the by-hand-approach would solve the issue. File was about 2GB of Data with 10 Million lines.)

Looking at the code:
```
void Main()
{
var csvConfig = new CsvConfiguration(CultureInfo.GetCultureInfo("en-us"))
            {
                NewLine = Environment.NewLine,
                PrepareHeaderForMatch = args => args.Header.Replace("_", "").ToLower() // CUSTOMER_ID -> customerid
            };

    using (var reader = new StreamReader("input.csv"))
    using (var csv = new CsvReader(reader, csvConfig))
    {
        var records = new List<Foo>();
        csv.Read();
        csv.ReadHeader(); 
```

It already breaks at the first ' csv.Read()'  just before the ' csv.ReadHeader()' with the errormessage dpenkova mentioned.

I am aware that our file is VERY big.

**Comment by Guacam-Ole** [comment]
Late answer to this, but in my case the reason for this was a wrong Newline-Setting. The CsvReader does not recognize the new line, so assumes the whole file is the Header. And tries to parse all 10GB in my case causing the issue.

Not sure if this is the same issue for @dpenkova but worth a look
